specification/communication/resource-manager/Microsoft.Communication/stable/2020-08-20/CommunicationService.json

```diff
@@ -0,0 +1,965 @@
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/CommunicationServiceName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The response describes CommunicationService access keys.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationServiceKeys"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "produces": [
+          "application/json"
+        ],
+        "x-ms-examples": {
+          "List keys": {
+            "$ref": "./examples/listKeys.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Communication/communicationServices/{communicationServiceName}/regenerateKey": {
+      "post": {
+        "description": "Regenerate CommunicationService access key. PrimaryKey and SecondaryKey cannot be regenerated at the same time.",
+        "operationId": "CommunicationService_RegenerateKey",
+        "summary": "Regenerate Key",
+        "tags": [
+          "CommunicationService"
+        ],
+        "parameters": [
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "Parameter that describes the Regenerate Key Operation.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RegenerateKeyParameters"
+            }
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/CommunicationServiceName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Created. The response contains the new key.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationServiceKeys"
+            }
+          },
+          "201": {
+            "description": "Created and an async operation is executing in background to make the new key take effect. The response contains new keys and an Azure-AsyncOperation header to query the async operation result.",
```

Comment:
> "description": "Created and an async operation is executing in background to make t [](start = 11, length = 84)

Is this async? Then please follow the rpc guideline. [https://github.com/Azure/azure-resource-manager-rpc/blob/f39512268c61b0a970766515d535a756495bb56d/v1.0/Addendum.md#call-action-post-asynchronously](https://github.com/Azure/azure-resource-manager-rpc/blob/f39512268c61b0a970766515d535a756495bb56d/v1.0/Addendum.md#call-action-post-asynchronously)

Reply:
RegenerateKey is not async, will remove the 201 from responses